### PR TITLE
Add mongodb

### PR DESCRIPTION
**Pull request by phlpphcknr**
- mongo db has been added
- the api token was changed from `@Value` to `@ConfigurationProperties`

### CODE REVIEW

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java [hunk 1]
```diff
@@ -1,34 +1,58 @@
...
 import com.hackner.musicompass.model.Artist;
 import com.hackner.musicompass.discogsapi.service.DiscogsArtistApiService;
 import lombok.Builder;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+
 import java.util.Optional;
 
 @Service
 @Builder
```
Comment (christophersiem) [inline backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:14]: no need to build the service class :)
```suggestion

```

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java [hunk 2]
```diff
@@ -1,34 +1,58 @@
...
                 .artistName(discogsArtist.getArtistName())
                 .artistImageUrl(discogsArtist.getArtistImageUrl())
-                .discogsId(discogsArtist.getDiscogsArtistId())
+                .discogsArtistId(discogsArtist.getDiscogsArtistId())
                 .discogsArtistUrl(discogsArtist.getDiscogsArtistUrl()).build());
     }
-}
\ No newline at end of file
+
+    public Artist getArtistByArtistName(String artistName) {
```
Comment (christophersiem) [inline backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:42]:
```suggestion
    public Artist getArtistByName(String artistName) {
```

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java [hunk 3]
```diff
@@ -1,34 +1,58 @@
...
+
+        DiscogsArtistSearchResults DiscogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistByArtistName(artistName);
```
Comment (christophersiem) [inline backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:44]:
```suggestion
        DiscogsArtistSearchResults DiscogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistByName(artistName);
```

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java [hunk 4]
```diff
@@ -1,34 +1,58 @@
...
-    public ArtistService(DiscogsArtistApiService discogsArtistApiService) {
+    public ArtistService(DiscogsArtistApiService discogsArtistApiService, ArtistMongoDb artistMongoDb) {
         this.discogsArtistApiService = discogsArtistApiService;
+        this.artistMongoDb = artistMongoDb;
     }
 
-    public Optional<Artist> getArtistByArtistName (String artistName) {
-        DiscogsArtistSearchResults DiscogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistByArtistName(artistName);
+    public Optional<Artist> getArtistBySearchTerm (String searchTerm) {
+        DiscogsArtistSearchResults DiscogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistByArtistName(searchTerm);
```
Comment (christophersiem) [inline backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:26]:
```suggestion
        DiscogsArtistSearchResults discogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistByArtistName(searchTerm);
```

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java [hunk 5]
```diff
@@ -1,34 +1,58 @@
...
         if (DiscogsArtistSearchResults.getResults().length == 0){
             return Optional.empty();
         }
-        DiscogsArtist discogsArtist = DiscogsArtistSearchResults.getResults()[0];
+
+        DiscogsArtist[] discogsArtistResultArray = DiscogsArtistSearchResults.getResults();
```
Comment (christophersiem) [inline backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:31]:
```suggestion
        DiscogsArtist[] discogsArtistResult = discogsArtistSearchResults.getResults();
```

**Review by christophersiem — APPROVED** [review]
looks good, only added some naming suggestions 🔥

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java (hunk 1 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:14]: thx. done.

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java (hunk 2 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:42]: yeah, a little redundant ... changed

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java (hunk 3 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:44]: same as above

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java (hunk 4 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:26]: thx

File: backend/src/main/java/com/hackner/musicompass/service/ArtistService.java (hunk 5 above)
Comment (phlpphcknr) [reply backend/src/main/java/com/hackner/musicompass/service/ArtistService.java:31]: this will be changed in the future anyway ....

**Pull request merged**